src/main/kotlin/com/openlattice/codex/CodexService.kt

```diff
@@ -0,0 +1,229 @@
+package com.openlattice.codex
+
+import com.auth0.json.mgmt.users.User
+import com.google.common.collect.ArrayListMultimap
+import com.google.common.collect.ListMultimap
+import com.hazelcast.core.HazelcastInstance
+import com.hazelcast.core.IMap
+import com.openlattice.apps.AppConfigKey
+import com.openlattice.apps.AppTypeSetting
+import com.openlattice.codex.controllers.CodexConstants
+import com.openlattice.controllers.exceptions.BadRequestException
+import com.openlattice.data.DataEdge
+import com.openlattice.data.DataGraphManager
+import com.openlattice.data.EntityDataKey
+import com.openlattice.data.EntityKeyIdService
+import com.openlattice.datastore.apps.services.AppService
+import com.openlattice.datastore.services.EdmManager
+import com.openlattice.edm.type.PropertyType
+import com.openlattice.hazelcast.HazelcastMap
+import com.openlattice.hazelcast.HazelcastQueue
+import com.openlattice.organizations.HazelcastOrganizationService
+import com.openlattice.organizations.roles.SecurePrincipalsManager
+import com.openlattice.twilio.TwilioConfiguration
+import com.twilio.Twilio
+import com.twilio.rest.api.v2010.account.Message
+import com.twilio.type.PhoneNumber
+import org.joda.time.DateTime
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import java.net.URI
+import java.time.Instant
+import java.time.OffsetDateTime
+import java.time.ZoneOffset.UTC
+import java.util.*
+import java.util.concurrent.Executors
+import java.util.stream.Stream
+import javax.servlet.http.HttpServletRequest
+
+@Service
+class CodexService(
```

Comment:
just wondering, why is this in datastore instead of cond-cli?

Reply:
because it's only scoped to datastore right now -- not called else